From 0302221752afe2074f9614d032fb5caf33f74115 Mon Sep 17 00:00:00 2001
From: Bin Meng <bmeng.cn@gmail.com>
Date: Wed, 6 Apr 2022 21:52:41 +0800
Subject: [PATCH 3/6] makefile: Fix the 'sgdisk' rule for mingw* builds

Currently the sgdisk build rule for mingw* builds is incorrect in
various aspects:

  * missing an object file gptcl.o for compiling and linking
  * linking to librpcrt4 library is missing too

With the fix we can enable the sgdisk build by default.

Signed-off-by: Bin Meng <bmeng.cn@gmail.com>
---
 Makefile.mingw64 | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile.mingw64 b/Makefile.mingw64
index 615b43b..3c63340 100644
--- a/Makefile.mingw64
+++ b/Makefile.mingw64
@@ -12,13 +12,13 @@ MBR_LIB_OBJS=$(MBR_LIBS:=.o)
 LIB_HEADERS=$(LIB_NAMES:=.h)
 DEPEND= makedepend $(CFLAGS)
 
-all:	gdisk fixparts
+all:	gdisk sgdisk fixparts
 
 gdisk:	$(LIB_OBJS) gdisk.o gpttext.o
 	$(CXX) $(LIB_OBJS) gdisk.o gpttext.o -lrpcrt4 -o gdisk64.exe
 
-sgdisk: $(LIB_OBJS) sgdisk.o
-	$(CXX) $(LIB_OBJS) sgdisk.o -lpopt -o sgdisk64.exe
+sgdisk: $(LIB_OBJS) sgdisk.o gptcl.o
+	$(CXX) $(LIB_OBJS) sgdisk.o gptcl.o -lrpcrt4 -lpopt -o sgdisk64.exe
 
 fixparts: $(MBR_LIB_OBJS) fixparts.o
 	$(CXX) $(MBR_LIB_OBJS) fixparts.o $(LDFLAGS) -o fixparts64.exe
-- 
2.35.1

